cartographer/mapping_3d/sparse_pose_graph.cc

```diff
@@ -423,6 +423,34 @@ void SparsePoseGraph::RunFinalOptimization() {
           .max_num_iterations());
 }
 
+void SparsePoseGraph::CalculateHistograms() {
+  translational_residual_ = common::Histogram();
+  for (const Constraint& constraint : constraints_) {
+    if (constraint.tag == Constraint::Tag::INTRA_SUBMAP) {
+      // in map frame
```

Comment:
Consistency: map <- node (since this is the transform from the node frame to the map frame)
Even better: Name the variables `optimized_node_to_map`, `node_to_submap_constraint`, `optimized_submap_to_map`.